Add Basic support for Microchip SAM D5X/E5X Family devices

ArunMCHP opened the pull request:
This PR adds initial support for Microchip’s SAM D5x/E5x family of SoCs, along with board support for the SAM E54 Xplained Pro development kit ([ATSAME54-XPRO](https://www.microchip.com/en-us/development-tool/atsame54-xpro)). It includes the basic SoC integration, devicetree files, and minimal configuration needed to bring up the board with a working UART console.
 
Associated pull request for the addition of the SAM D5x/E5x devices to hal_microchip:
https://github.com/zephyrproject-rtos/hal_microchip/pull/33

The naming convention of driver files and device organization is as per the RFC : https://github.com/zephyrproject-rtos/zephyr/issues/92168

ArunMCHP committed:
soc: microchip: add support for SAM D5x/E5x SoC series

Adds initial SoC-level support for the Microchip SAM D5x
and E5x series, including SoC definition files.

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
boards: microchip: add SAM E54 Xplained Pro board support

Add initial support for the Microchip SAM E54 Xplained Pro board
(ATSAME54-XPRO).

Product page: https://www.microchip.com/en-us/development-tool/atsame54-xpro

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
dts: arm: microchip: add dtsi files for Microchip SAM D5x/E5x SoC series

Adds common and SoC-specific .dtsi files for the Microchip
SAM D5x/E5x family. These files define core peripherals,
address maps, and interrupt controller structure shared
across the SAM D5x/E5x variants.

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
west.yml: Update west.yml for SAM_D5X_E5X dfp

Update west.yml for SAM_D5X_E5X Device family pack

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

sunil-abraham committed:
dts: arm: microchip: add clock dts node and bindings for SAM D5x/E5x

Add clock dts node and minimal set of binding parameters
for clock_control driver.

Signed-off-by: Sunil Abraham <sunil.abraham@microchip.com>

sunil-abraham committed:
drivers: clock_control: microchip: clock support for SAM D5x/E5x

Add clock control driver with minimal functionality.
Implement basic on, off, get_status and get_rate API.

Signed-off-by: Sunil Abraham <sunil.abraham@microchip.com>

AzharMCHP committed:
dts: arm: microchip: add pinctrl dts node and bindings for Port G1 IP

Add pinctrl dts node and binding parameters for Microchip
Pinctrl Port G1 IP

Signed-off-by: Mohamed Azhar <mohamed.azhar@microchip.com>

sunil-abraham committed:
dts: arm: microchip: add uart dts node and bindings for sercom g1 IPs

Add uart dts nodes and minimal set of binding parameters
for sercom uart driver.

Signed-off-by: Sunil Abraham <sunil.abraham@microchip.com>

sunil-abraham committed:
drivers: uart: microchip: add minimal support for sercom uart g1 IPs

Add uart driver with minimal features.
Implement polling receive and transmit functionality.

Signed-off-by: Sunil Abraham <sunil.abraham@microchip.com>

AzharMCHP committed:
drivers: pinctrl: microchip: add pinctrl driver for Port G1 IP

Add pinctrl driver for Microchip Port G1 Peripheral IPs

Signed-off-by: Mohamed Azhar <mohamed.azhar@microchip.com>

nordicjm commented on soc/microchip/sam/sam_d5x_e5x/Kconfig.soc:
```diff
@@ -0,0 +1,20 @@
+config SOC_FAMILY_MCHP_SAM_D5X_E5X
+	bool
+
+config SOC_FAMILY
+	default "microchip_sam_d5x_e5x" if SOC_FAMILY_MCHP_SAM_D5X_E5X
```
names do not match

nandojve replied:
Hi @nordicjm ,

It is not clear what naming you are referring for:

Is the `default "microchip_sam_d5x_e5x"` that should match with the folder structure ?

`soc/microchip/sam/sam_d5x_e5x` or `soc/microchip/sam/d5x_e5x`
or even

`SOC_FAMILY_MCHP_SAM_D5X_E5X` should be `SOC_FAMILY_MICROCHIP_SAM_D5X_E5X` ?

nordicjm replied:
`soc.yml` values e.g family must match the string here and Kconfig name for the family

ArunMCHP replied:
Hi @nordicjm, @nandojve ,

Thank you both for the clarification.

I have now updated the `soc.yml` and ensured that the `Kconfig` symbols and values are aligned accordingly. Please let me know if anything else needs to be addressed.

nordicjm commented on soc/microchip/sam/sam_d5x_e5x/samd51/Kconfig.soc:
```diff
@@ -0,0 +1,60 @@
+	help
+	  Enable support for Microchip SAMD51 Cortex-M4F microcontrollers.
+
+config SOC_SERIES
+	default "samd51" if SOC_SERIES_MCHP_SAMD51
```
does not match

nordicjm commented on soc/microchip/sam/sam_d5x_e5x/samd51/Kconfig.soc:
```diff
@@ -0,0 +1,60 @@
+
+config SOC_SERIES
+	default "samd51" if SOC_SERIES_MCHP_SAMD51
+
```
```suggestion
```

ArunMCHP replied:
fixed.

nordicjm commented on soc/microchip/sam/sam_d5x_e5x/samd51/Kconfig.soc:
```diff
@@ -0,0 +1,60 @@
+config SOC_ATSAMD51P20A
+	bool
+	select SOC_SERIES_MCHP_SAMD51
+
```
```suggestion
```

nordicjm commented on soc/microchip/sam/sam_d5x_e5x/same51/Kconfig.soc:
```diff
@@ -0,0 +1,50 @@
+	select SOC_FAMILY_MCHP_SAM_D5X_E5X
+	help
+	  Enable support for Microchip SAME51 Cortex-M4F microcontrollers.
+
```
```suggestion
```
fix this issue in the whole PR

nordicjm commented on soc/microchip/sam/sam_d5x_e5x/same51/Kconfig.soc:
```diff
@@ -0,0 +1,50 @@
+	  Enable support for Microchip SAME51 Cortex-M4F microcontrollers.
+
+
+config SOC_SERIES
+	default "same51" if SOC_SERIES_MCHP_SAME51
```
does not match, fix issue in whole PR

nordicjm commented on drivers/clock_control/Kconfig.mchp:
```diff
@@ -0,0 +1,30 @@
+	default y
+	help
+	  This option enables get rate API.
+
```
```suggestion
```

ArunMCHP replied:
Fixed.

NhMchp replied:
Yes; still the empty space is not removed

ArunMCHP replied:
Remove the extra empty line.

nordicjm commented on drivers/pinctrl/Kconfig.mchp:
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2025 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config PINCTRL_MCHP_PORT_ID_U2210_2_2_0
+	bool
+	default n
+	help
+	  Internal flag to enable PORT ID U2210_2_2_0 support.
```
put this in an if so it's not bleeding into other devices

ArunMCHP replied:
We moved this symbol to family kconfig.soc, and kept inside if SOC_FAMILY_MCHP_SAM_D5X_E5X.
path : soc/microchip/sam/sam_d5x_e5x/Kconfig.soc

nordicjm commented on drivers/pinctrl/Kconfig.mchp:
```diff
@@ -0,0 +1,16 @@
+
+config PINCTRL_MCHP_PORT_ID_U2210_2_2_0
+	bool
+	default n
```
```suggestion
```
fix in whole PR

nordicjm commented on drivers/serial/Kconfig.mchp:
```diff
@@ -0,0 +1,18 @@
+	select PINCTRL
+	select UART_MCHP_SERCOM_ID_U2201_5_0_0 if SOC_FAMILY_MCHP_SAM_D5X_E5X
+	help
+		This option enables the USART driver for Microchip sercom-g1 IPs.
```
help text indent in 1x tab followed by 2x spaces

ArunMCHP replied:
Fixed.

nordicjm commented on boards/microchip/sam/sam_e54_xpro/doc/img/atsam_e54_xpro.webp:
reduce resolution and put through https://tinypng.com/

nordicjm commented on boards/microchip/sam/sam_e54_xpro/doc/img/atsam_e54_xpro_pinout.webp:
reduce resolution and put through https://tinypng.com/

ArunMCHP replied:
Reduced resolution using https://tinypng.com/ and updated.

nordicjm replied:
these images combined are 135KiB, they should be smaller

parthitce reviewed (changes requested):
Please re-order the commits.

ArunMCHP commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x.dtsi:
Thanks, @parthitce ! That’s a good idea, we’ve reordered the commits and moved the final DTS into a single commit as you suggested.

nandojve replied:
The SoC DTS must be before SoC. Why ? There is usually a dependency order that must be take in consideration.
BTW, keep the dts with only the nodes that are really necessary. Let nodes from the drivers in their own commit together with the bindings.

We need to establish the correct order to avoid bisect issues in the future.

ArunMCHP commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x.dtsi:
```diff
@@ -0,0 +1,54 @@
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv7m-mpu";
+				reg = <0xE000ED90 0x2c>;
```
Updated.

ArunMCHP commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x.dtsi:
```diff
@@ -0,0 +1,54 @@
+		};
+
+		nvmctrl: nvmctrl@41004000 {
+			compatible = "microchip,nvmctrl-g1";
```
Removed the node.

NhMchp commented on dts/bindings/clock/microchip,sam-d5x-e5x-clock.yaml:
```diff
@@ -0,0 +1,18 @@
+  - name: pinctrl-device.yaml
+
+description: Binding for Microchip SAM_D5x_E5x SOC family clock (OSCCTRL, OSC32KCTRL, GCLK, MCLK)
+compatible: "microchip,sam-d5x-e5x-clock"
```
@parthitce, Exceptions may apply to certain peripherals, such as clocks, where the implementation involves multiple hardware IPs and is specific to a device family. In these cases, when grouping drivers, it may be necessary to create a separate driver for each device family. Therefore, it is preferable to name the driver after the device family rather than using a grouping number. I have updated the RFC to reflect this exception.

ArunMCHP commented on drivers/clock_control/clock_mchp_sam_d5x_e5x.c:
```diff
@@ -0,0 +1,730 @@
+	default:
+		LOG_ERR("Unsupported mclkbus");
+		break;
+	}
```
New line provided.

ArunMCHP commented on drivers/clock_control/clock_mchp_sam_d5x_e5x.c:
```diff
@@ -0,0 +1,730 @@
+	bool is_wait = false;
+	uint32_t on_timeout_ms = 0;
+
+	subsys.val = (uint32_t)sys;
```
Fixed.

sunil-abraham commented on drivers/clock_control/clock_mchp_sam_d5x_e5x.c:
```diff
@@ -0,0 +1,730 @@
+	do {
+		/* Validate subsystem. */
+		if (CLOCK_SUCCESS != clock_check_subsys(subsys)) {
+			break;
```
The do-while block is utilized to prevent multiple return statements within this function. Additional content will be incorporated into this block once the full-scale clock driver is implemented.

parthitce replied:
NACK! Sanity checks or error cases eventually needs return. This whole file is full of do while without technical reasons.

NhMchp replied:
> NACK! Sanity checks or error cases eventually needs return. This whole file is full of do while without technical reasons.

@parthitce , we follow MISRA C rule 15.5 which says "A function should have a single point of exit at the end". There are multiple ways to do it. As 'goto' is generally discouraged, one of the option is to embed the code inside do-while(0). We are maintaining this as our internal coding guideline. Do you see any issues/conflicts for following this guideline with Zephyr coding ?

sunil-abraham commented on soc/microchip/sam/sam_d5x_e5x/common/soc.c:
```diff
@@ -0,0 +1,12 @@
+
+void soc_reset_hook(void)
+{
+	/* Intentionally empty */
```
The soc_reset_hook has been included to maintain a consistent structure. Although this file is currently empty, it serves as a placeholder for future peripheral driver additions and enables users to add custom code to execute during reset.

parthitce commented on drivers/clock_control/clock_mchp_sam_d5x_e5x.c:
```diff
@@ -0,0 +1,709 @@
+		gclkgen_src = (gclk_regs->GCLK_GENCTRL[gclkgen_id] & GCLK_GENCTRL_SRC_Msk) >>
+			      GCLK_GENCTRL_SRC_Pos;
+		if (gclkgen_called_src == gclkgen_src) {
+			ret_val = -ELOOP;
```
ELOOP? Have you read `clock_control_get_rate` ?

sunil-abraham replied:
Removed ELOOP, and the API now return ENOTSUP to exactly match the spec in clock_control.h.

nordicjm commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x_j.dtsi:
```diff
@@ -0,0 +1,11 @@
+
+/ {
+	soc {
+		/* intentionally empty */
```
add it when it is populated later, not just to add something empty

ArunMCHP replied:
@nordicjm Empty files are removed. Thanks.

ArunMCHP commented:
@parthitce Thanks for your valuable comments and suggestions.
 
 Updated the PR as per your comments :
 
* Reordered the commits as per your suggestion.
* Removed empty files.
* Corrected CMakeLists.txt.
* Fixed camel case issues in DTS files.
* Updated clock source files based on your comments.
* Moved DTSI files into a single commit.
* And provided comments for do-while usage, ELOOP and IP version Kconfig symbols.

ArunMCHP commented:
Thank you, @nandojve, for the guidance.

The commits have now been reordered, SoC DTSI files are added before the SoC, and the individual drivers update their respective DTSI nodes in separate commits as suggested.
Please let us know if any further adjustments are needed.

nandojve reviewed (changes requested):
Hi @ArunMCHP @sunil-abraham,

Overall LGTM, nice work!

I made a in deep review and found a few more things.

ArunMCHP commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x_19.dtsi:
```diff
@@ -0,0 +1,17 @@
+
+/ {
+	soc {
+		sram0: memory@20000000 {
```
Sorted.

ArunMCHP commented on soc/microchip/sam/sam_d5x_e5x/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+zephyr_include_directories_ifdef(CONFIG_SOC_SERIES_MCHP_SAMD51 samd51)
+zephyr_include_directories_ifdef(CONFIG_SOC_SERIES_MCHP_SAME51 same51)
+zephyr_include_directories_ifdef(CONFIG_SOC_SERIES_MCHP_SAME53 same53)
+zephyr_include_directories_ifdef(CONFIG_SOC_SERIES_MCHP_SAME54 same54)
```
Updated as per the comment. Thanks.

ArunMCHP commented on soc/microchip/sam/sam_d5x_e5x/Kconfig.soc:
```diff
@@ -0,0 +1,27 @@
+	default "mchp_sam_d5x_e5x" if SOC_FAMILY_MCHP_SAM_D5X_E5X
+
+# List of Peripheral IPs available in SOC_FAMILY_MCHP_SAM_D5X_E5X family
+if SOC_FAMILY_MCHP_SAM_D5X_E5X
```
Moved to later commits, where it is applicable.

ArunMCHP commented on soc/microchip/sam/sam_d5x_e5x/Kconfig.soc:
```diff
@@ -0,0 +1,27 @@
+# List of Peripheral IPs available in SOC_FAMILY_MCHP_SAM_D5X_E5X family
+if SOC_FAMILY_MCHP_SAM_D5X_E5X
+
+config UART_MCHP_SERCOM_U2201_5_0_0
```
Yes, Moved to UART commit.

ArunMCHP commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x.dtsi:
```diff
@@ -35,6 +35,23 @@
+			compatible = "microchip,sam-d5x-e5x-clock";
+			reg = <0x40000800 0x24>, <0x40001000 0x58>,
+				  <0x40001400 0x20>, <0x40001c00 0x140>;
+			reg-names = "mclk", "oscctrl", "osc32kctrl", "gclk";
```
Corrected.

ArunMCHP commented on dts/bindings/clock/microchip,sam-d5x-e5x-gclkperiph.yaml:
```diff
@@ -0,0 +1,21 @@
+    type: int
+    description: |
+      The subsystem cell is to identify a clock controller sub-system.
+      The subsystem can be referred from include\zephyr\dt-bindings\clock\mchp_sam_d5x_e5x_clock.h,
```
Corrected.

ArunMCHP commented on dts/bindings/clock/microchip,sam-d5x-e5x-mclkperiph.yaml:
```diff
@@ -0,0 +1,21 @@
+    type: int
+    description: |
+      The subsystem cell is to identify a clock controller sub-system.
+      The subsystem can be referred from include\zephyr\dt-bindings\clock\mchp_sam_d5x_e5x_clock.h,
```
Corrected.

ArunMCHP commented on drivers/clock_control/CMakeLists.txt:
```diff
@@ -10,6 +10,7 @@ zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_FIXED_RATE_CLOCK    clock_cont
 zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_GD32                clock_control_gd32.c)
 zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_LITEX               clock_control_litex.c)
 zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_LPC11U6X            clock_control_lpc11u6x.c)
+zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_MCHP_SAM_D5X_E5X    clock_mchp_sam_d5x_e5x.c)
```
File renamed and updated CMakeList.txt.

ArunMCHP commented on drivers/pinctrl/CMakeLists.txt:
```diff
@@ -57,5 +57,6 @@ zephyr_library_sources_ifdef(CONFIG_PINCTRL_SY1XX pinctrl_sy1xx.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_REALTEK_RTS5912 pinctrl_realtek_rts5912.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_WCH_20X_30X_AFIO pinctrl_wch_20x_30x_afio.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_WCH_00X_AFIO pinctrl_wch_00x_afio.c)
+zephyr_library_sources_ifdef(CONFIG_PINCTRL_MCHP_PORT_G1 pinctrl_mchp_port_g1.c)
```
The file was not originally in sorted order. However, it has now been moved to proper location.

ArunMCHP commented on drivers/pinctrl/Kconfig:
```diff
@@ -82,6 +82,7 @@ source "drivers/pinctrl/Kconfig.sy1xx"
 source "drivers/pinctrl/Kconfig.realtek_rts5912"
 source "drivers/pinctrl/Kconfig.wch_20x_30x_afio"
 source "drivers/pinctrl/Kconfig.wch_00x_afio"
+source "drivers/pinctrl/Kconfig.mchp"
```
The file was not originally in sorted order. However, it has now been moved to proper location.

ArunMCHP commented on dts/bindings/arm/microchip,sercom-g1.yaml:
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2025 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: Microchip SERCOM multi-protocol (UART, SPI, I2C) SERCOM unit, group g1
```
Reviewed all bindings and updated all as per the comment.

ArunMCHP commented on drivers/serial/uart_mchp_sercom_g1.c:
```diff
@@ -0,0 +1,768 @@
+		}
+		uart_wait_sync(regs, clock_external);
+	} while (0);
+	return retval;
```
Corrected.

ArunMCHP commented on boards/microchip/sam/sam_e54_xpro/sam_e54_xpro-pinctrl.dtsi:
```diff
@@ -0,0 +1,16 @@
+&pinctrl {
+	sercom2_uart_default: sercom2_uart_default {
+		group1 {
+			pinmux = <PB25D_SERCOM2_PAD0>,
+					 <PB24D_SERCOM2_PAD1>;
```
Updated.

nandojve replied:
Still not OK

ArunMCHP commented on boards/microchip/sam/sam_e54_xpro/sam_e54_xpro.yaml:
```diff
@@ -0,0 +1,18 @@
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
Not using, dropped them.

sunil-abraham commented on drivers/serial/uart_mchp_sercom_g1.c:
```diff
@@ -0,0 +1,768 @@
+	do {
+		if (uart_is_rx_complete(regs, clock_external) == false) {
+			retval = -EBUSY;
+			break;
```
Updated

sunil-abraham commented on drivers/clock_control/clock_mchp_sam_d5x_e5x.c:
```diff
@@ -0,0 +1,709 @@
+#define DT_DRV_COMPAT microchip_sam_d5x_e5x_clock
+LOG_MODULE_REGISTER(clock_mchp_sam_d5x_e5x, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#define CLOCK_NODE DT_NODELABEL(clock)
```
Shall update on the full scale clock driver.

nandojve commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x.dtsi:
```diff
@@ -34,6 +35,24 @@
+
+		clock: clock@40000800 {
+			compatible = "microchip,sam-d5x-e5x-clock";
+			reg = <0x40000800 0x24>, <0x40001000 0x58>,
+				  <0x40001400 0x20>, <0x40001c00 0x140>;
+			reg-names = "mclk", "oscctrl",
+						"osc32kctrl", "gclk";
```
Alignment, remember that is tab 8 most you can then complete with spaces.

```diff
			reg = <0x40000800 0x24>, <0x40001000 0x58>,
-                                 <0x40001400 0x20>, <0x40001c00 0x140>;
+                             <0x40001400 0x20>, <0x40001c00 0x140>;
                        reg-names = "mclk", "oscctrl",
-                                               "osc32kctrl", "gclk";
+                                   "osc32kctrl", "gclk";
```

ArunMCHP replied:
Thanks @nandojve, Corrected now.

nandojve commented on drivers/pinctrl/pinctrl_mchp_port_g1.c:
```diff
@@ -0,0 +1,185 @@
+ *
+ * @param nodelabel The node label to check in the device tree.
+ */
+#define MCHP_PORT_ADDR_OR_NONE(nodelabel)                                                          \
+	IF_ENABLED(DT_NODE_EXISTS(DT_NODELABEL(nodelabel)),            \
+		(DT_REG_ADDR(DT_NODELABEL(nodelabel))))
```
Still open

AzharMCHP replied:
The spacing before the backslash (\) in the macro definition line is automatically adjusted by clang-format.

ArunMCHP replied:
Updated.

nandojve commented on dts/arm/microchip/sam/sam_d5x_e5x/common/samd5xe5x.dtsi:
```diff
@@ -54,6 +54,26 @@
+			compatible = "microchip,sercom-g1";
+			status = "disabled";
+			reg = <0x40003000 0x31>;
+			interrupts = <46 0>, <47 0>, <48 0>, <49 0>;
+			clocks = <&mclkperiph CLOCK_MCHP_MCLKPERIPH_ID_APBA_SERCOM0>,
+					 <&gclkperiph CLOCK_MCHP_GCLKPERIPH_ID_SERCOM0_CORE>;
+			clock-names = "mclk", "gclk";
```
Alignment, review all sercom nodes.
```diff
                        clocks = <&mclkperiph CLOCK_MCHP_MCLKPERIPH_ID_APBA_SERCOM0>,
-                                        <&gclkperiph CLOCK_MCHP_GCLKPERIPH_ID_SERCOM0_CORE>;
+                                <&gclkperiph CLOCK_MCHP_GCLKPERIPH_ID_SERCOM0_CORE>;
                        clock-names = "mclk", "gclk";
```

ArunMCHP replied:
Corrected.

nandojve reviewed (changes requested):
I found small things now.

nandojve commented on drivers/serial/uart_mchp_sercom_g1.c:
```diff
@@ -0,0 +1,768 @@
+			}
+		}
+		uart_wait_sync(regs, clock_external);
+	} while (0);
```
Ahh, understood!
In this case, please, go ahead!

nandojve reviewed (dismissed):
Hi MCHP team,

Everything is fine from my side.
Thank you for all your effort to make this possible.

ArunMCHP commented:
Hi @nandojve,
Thank you for your time and valuable comments throughout the review. Your feedback really helped improve the quality of the work.
Appreciate your support!

ArunMCHP commented:
@nandojve 
https://github.com/zephyrproject-rtos/zephyr/pull/93450#discussion_r2249903423 is resolved now.

nordicjm commented on drivers/clock_control/Kconfig.mchp:
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2025 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Symbol for enabling clock driver for a family
+config CLOCK_CONTROL_MCHP_SAM_D5X_E5X
+	bool
+	default y
+	depends on DT_HAS_MICROCHIP_SAM_D5X_E5X_CLOCK_ENABLED
+	help
+	  This symbol define clock driver for Microchip SAM_D5X_E5X family.
+
+# Define menu symbol for configuring clock features
+menuconfig CLOCK_CONTROL_MCHP
+	bool "Microchip clock controller driver"
+	default y
+	depends on CLOCK_CONTROL_MCHP_SAM_D5X_E5X
+	help
+	  Enable support for Microchip clock controller driver.
```
this is superfluous
```suggestion
menuconfig CLOCK_CONTROL_MCHP
	bool "Microchip clock controller driver"
	default y
	depends on DT_HAS_MICROCHIP_SAM_D5X_E5X_CLOCK_ENABLED
	help
	  Enable support for Microchip clock controller driver.
```

nordicjm replied:
can have symbols like `CLOCK_CONTROL_MCHP_SAM_D5X_E5X` (which has the depends on DT_...) select a Kconfig with no prompt e.g. `CLOCK_CONTROL_MCHP_COMMON` then have options displayed there

nordicjm replied:
```suggestion
config CLOCK_CONTROL_MCHP_COMMON
        bool
        help
          Common option for microchip clock control drivers

config CLOCK_CONTROL_MCHP_SAM_D5X_E5X
        bool "Text here"
        depends on DT_...
        select CLOCK_CONTROL_MCHP_COMMON
```

nordicjm commented on boards/microchip/sam/sam_e54_xpro/doc/index.rst:
```diff
@@ -0,0 +1,160 @@
+
+.. zephyr:board-supported-hw::
+
+The default configuration can be found in the Kconfig
+:zephyr_file:`boards/microchip/sam/sam_e54_xpro/sam_e54_xpro_defconfig`.
+
```
```suggestion
```

ArunMCHP commented:
Hi @nordicjm,
Thanks for your follow-up review. We’ve addressed the latest comments you provided. Could you please take another look and approve the PR if everything looks good?

nordicjm reviewed (dismissed):
soc/board files OK

nandojve commented:
Hi @ArunMCHP ,

I think as soon @parthitce approves you can ask to MCHP merge the HAL content and update the hash here.

ArunMCHP commented:
Hi @nandojve,

Sure, I’ll coordinate with the MCHP maintainers to get the HAL PR merged and will update the hash in the west.yml file. Thanks!

ArunMCHP commented:
Hi @nandojve  @nordicjm @parthitce,

We’ve updated the PR with the latest commit hash from [hal_microchip ](https://github.com/zephyrproject-rtos/hal_microchip/commits/master/)and rebased it onto the current main branch to resolve conflicts in the west.yml file.

Kindly request you to review and approve the PR again at your convenience.
Thank you for your continued support!

parthitce reviewed (dismissed):
Thanks for your efforts!

ArunMCHP commented:
Hi @jamie, @parthitce, @nandojve,

Thank you all for reviewing and approving the PR!

Kindly let us know the next steps. If everything looks good, can we proceed with merging?

Thanks again for your support!

kartben reviewed (changes requested):
Some minor problems spotted in the documentation page of the new board

Thanks!

kartben reviewed (commented):
Also note that you may want to refrain from using the word "binding" in the description of the bindings you're introducing, and may want to look at adding a concise `title:` property which is what will get picked up for the autogenerated list of supported hardware features 
https://docs.zephyrproject.org/latest/boards/microchip/sam/sama7g54_ek/doc/index.html#supported-features

ArunMCHP commented on boards/microchip/sam/sam_e54_xpro/doc/index.rst:
```diff
@@ -0,0 +1,157 @@
+
+.. image:: img/atsam_e54_xpro.webp
+     :align: center
+     :alt: SAM_E54_XPRO
```
Removed.

ArunMCHP commented on boards/microchip/sam/sam_e54_xpro/doc/index.rst:
```diff
@@ -0,0 +1,157 @@
+
+.. image:: img/atsam_e54_xpro_pinout.webp
+     :align: center
+     :alt: SAME54-XPRO-pinout
```
Removed.

ArunMCHP commented on boards/microchip/sam/sam_e54_xpro/doc/index.rst:
```diff
@@ -0,0 +1,157 @@
+- SERCOM7 I2C SDA  : PD08
+- SERCOM7 I2C SCL  : PD09
+- USB DP           : PA25
+- USB DM           : PA24
```
Thanks @kartben, We have removed this section and given reference to board user guide.

ArunMCHP commented:
> Also note that you may want to refrain from using the word "binding" in the description of the bindings you're introducing, and may want to look at adding a concise `title:` property which is what will get picked up for the autogenerated list of supported hardware features https://docs.zephyrproject.org/latest/boards/microchip/sam/sama7g54_ek/doc/index.html#supported-features

Thanks @kartben. We have added a title to each binding we introduced and also removed the word "binding" from the description section. 

We have addressed all the review comments,  kindly review and merge the PR if everything looks good.

kartben commented on boards/microchip/sam/sam_e54_xpro/doc/index.rst:
```diff
@@ -0,0 +1,97 @@
+which provides both a standard debug interface and a virtual serial port used as the Zephyr console.
+SERCOM2 of the ATSAME54P20A MCU is configured as a USART for console output.
+
+Follow these steps to program and debug the board:
+==================================================
+
```
This seems unnecessary, you're already in the Programming and Debugging section


```suggestion
```

ArunMCHP commented:
Hi @kartben,
Thank you for your valuable feedback. We’ve addressed your previous comment, kindly re-review and approve the PR when convenient.

Hi @parthitce, @nordicjm , @nandojve,
Thanks again for your earlier reviews and approvals. Since we’ve made documentation updates based on @kartben comments, we kindly request your re-approval on the updated PR.
